src/llfs/volume.test.cpp

```diff
@@ -2079,4 +2118,345 @@ TEST_F(VolumeTest, AppendableJobGrantSize)
+  // post_hatl values respectively. .
+
+  LOG(INFO) << "Starting DeadPageRefCountVariant test for " << max_seeds << " iterations...";
+
+  auto main_test_block = [&]() {
+    for (u32 current_seed = 0; current_seed < max_seeds; ++current_seed) {
+      LOG_EVERY_N(INFO, 100) << BATT_INSPECT(current_seed) << BATT_INSPECT(max_seeds);
+      ASSERT_NO_FATAL_FAILURE(
+          this->run_dead_page_recovery_test_variant(current_seed, current_seed, current_seed));
+    }
+  };
+  // Note that we are enabling thread-safe mode for Death-Test.
+  //
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+  EXPECT_EXIT(main_test_block(), testing::ExitedWithCode(6), "FATAL.*");
+  ::testing::FLAGS_gtest_death_test_style = "fast";
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+void VolumeSimTest::alloc_one_page_and_commit(RecoverySimState& state, llfs::StorageSimulation& sim,
+                                              llfs::Volume& volume, llfs::SlotRange& slot_offset)
+{
+  std::unique_ptr<llfs::PageCacheJob> job = volume.new_job();
+
+  ASSERT_NE(job, nullptr);
+
+  batt::StatusOr<llfs::PageId> new_page_id =
+      this->build_page_with_refs_to({}, llfs::PageSize{1 * kKiB}, *job, sim);
+
+  ASSERT_TRUE(new_page_id.ok()) << BATT_INSPECT(new_page_id.status());
+
+  // Save the page_id so we can use it later.
+  //
+  state.first_page_id = *new_page_id;
+
+  // Write the page and slot to the Volume.
+  //
+  batt::StatusOr<llfs::SlotRange> slot_range =
+      this->commit_job_to_root_log(std::move(job), *new_page_id, volume, sim);
+
+  ASSERT_TRUE(slot_range.ok()) << BATT_INSPECT(slot_range.status());
+
+  sim.log_event("first job successfully appended! slot_range=", *slot_range);
+  slot_offset = *slot_range;
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+auto VolumeSimTest::RecoverySimState::get_slot_visitor_for_dead_pages()
+{
+  return [this](const llfs::SlotParse /*slot*/, const llfs::PageId& page_id) {
+    if (page_id == this->first_page_id) {
+      this->recovered_first_page = true;
+    } else {
+      this->no_unknown_pages = false;
+    }
+    return batt::OkStatus();
+  };
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+void display_metric_data(llfs::Volume& volume)
+{
+  LLFS_VLOG(1) << "PageRecycler metrics: "
+               << BATT_INSPECT(volume.page_recycler_metrics().insert_count);
+  LLFS_VLOG(1) << "PageRecycler metrics: "
+               << BATT_INSPECT(volume.page_recycler_metrics().remove_count);
+  LLFS_VLOG(1) << "PageRecycler metrics: "
+               << BATT_INSPECT(volume.page_recycler_metrics().page_drop_ok_count);
+  LLFS_VLOG(1) << "PageRecycler metrics: "
+               << BATT_INSPECT(volume.page_recycler_metrics().page_drop_error_count);
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+void batt_task_yield(u32 count)
+{
+  for (uint32_t idx = 0; idx < count; idx++) {
```

Comment:
nit: Please use `i` or `index`.

Reply:
I will change this to avoid adding any new variable.